apps/indexer/src/api/repositories/drizzle.ts

```diff
@@ -156,6 +175,136 @@ export class DrizzleRepository {
     return db.$count(proposalsOnchain);
   }
 
+  async getProposalNonVoters(
+    proposalId: string,
+    skip: number,
+    limit: number,
+    orderDirection: "asc" | "desc",
+    addresses?: Address[],
+  ): Promise<{ voter: Address; votingPower: bigint }[]> {
+    return await db
+      .select({
+        voter: accountPower.accountId,
+        votingPower: accountPower.votingPower,
+      })
+      .from(accountPower)
+      .leftJoin(
+        votesOnchain,
+        and(
+          eq(votesOnchain.proposalId, proposalId),
+          eq(votesOnchain.voterAccountId, accountPower.accountId),
+        ),
+      )
+      .where(
+        and(
+          ...(addresses ? [inArray(accountPower.accountId, addresses)] : []),
+          gt(accountPower.votingPower, 0n),
+          isNull(votesOnchain.proposalId), // NULL means they didn't vote on this proposal
+        ),
+      )
+      .orderBy(
+        orderDirection === "asc"
+          ? asc(accountPower.votingPower)
+          : desc(accountPower.votingPower),
+      )
+      .limit(limit)
+      .offset(skip);
+  }
+
+  async getProposalNonVotersCount(proposalId: string): Promise<number> {
+    const countResult = await db
+      .select({ count: count(accountPower.accountId) })
+      .from(accountPower)
+      .leftJoin(
+        votesOnchain,
+        and(
+          eq(votesOnchain.proposalId, proposalId),
+          eq(votesOnchain.voterAccountId, accountPower.accountId),
+        ),
+      )
+      .where(
+        and(gt(accountPower.votingPower, 0n), isNull(votesOnchain.proposalId)),
+      );
+    return countResult[0]?.count || 0;
+  }
+
+  async getLastVotersTimestamp(
+    voters: Address[],
+  ): Promise<Record<Address, bigint>> {
+    const timestamps = await db
+      .select({
+        voterAccountId: votesOnchain.voterAccountId,
+        lastVoteTimestamp: max(votesOnchain.timestamp),
+      })
+      .from(votesOnchain)
+      .where(inArray(votesOnchain.voterAccountId, voters))
+      .groupBy(votesOnchain.voterAccountId)
+      .orderBy(desc(max(votesOnchain.timestamp)));
```

Comment:
```suggestion
```

i think we dont need this operation. Like, this func is used on getProposalNonVoters. Over there we already order.